recipes-kernel/linux/linux-intel-rt-acrn-uos_4.19.bb

```diff
@@ -32,6 +32,7 @@ KERNEL_FEATURES_append = "features/netfilter/netfilter.scc \
                           features/security/security.scc  \
                           cfg/hv-guest.scc \
                           cfg/paravirt_kvm.scc \
+                          features/tpm/tpm.scc \
```

Comment:
I think, everyone would not want to enable TPM. We should enable TPM based on distro feature.

${@bb.utils.contains("DISTRO_FEATURES", "tpm", " features/tpm/tpm.scc", "" ,d)}

It can be enabled in local.conf by adding

DISTRO_FEATURES_append = " tpm"

Reply:
Making tpm as MACHINE_FEATURES would enable it for ESE kernel too.

Reply:
> I think, everyone would not want to enable TPM. We should enable TPM based on machine feature.
> 
> ${@bb.utils.contains("MACHINE_FEATURES", "tpm", " features/tpm/tpm.scc", "" ,d)}
> 
> It can be enabled in local.conf by adding
> 
> MACHINE_FEATURES_append = " tpm"

You mean: not everyone would want to enable TPM if there's a TPM device for a RTVM ? I don't agree with you. For RTVM, if there's a TPM device, we would use it. If there's not a TPM device, enable these Kconfig setting is OK too.

Reply:
I just checked that, TPM configuration is already enabled with common bsp configuration. 

http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/features/tpm/tpm.cfg?h=yocto-5.4
http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/bsp/intel-common/intel-common-drivers.scc?h=yocto-5.4#n60

Please check at runtime ?

Reply:
Thanks @cheeyanglee @saininav ‘s comments, we will check later

Reply:
> I just checked that, TPM configuration is already enabled with common bsp configuration.
> 
> http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/features/tpm/tpm.cfg?h=yocto-5.4
> http://git.yoctoproject.org/cgit/cgit.cgi/yocto-kernel-cache/tree/bsp/intel-common/intel-common-drivers.scc?h=yocto-5.4#n60
> 
> Please check at runtime ?

Hi Saininav
Could we don't enable TPM configure for common kernel?  We only want to enable this setting for RT kernel.

Reply:
Is enabling TPM config causing any harm for common kernel ?

Reply:
On native, I think not. On virtualization, it may will. For example, if we enable TPM on SOS, if there's a TPM device on this board, the SOS may use it. However, if we want to boot a RT UOS. The RT UOS will need TPM for security. In this case, we may can't unbind the TPM from SOS and pass it through to RT UOS.

Reply:
> Thanks @cheeyanglee @saininav ‘s comments, we will check later

Have confirmed that It's OK now, thanks

Reply:
@saininav do you know when enabled TPM configure for common kernel?

Reply:
> On native, I think not. On virtualization, it may will. For example, if we enable TPM on SOS, if there's a TPM device on this board, the SOS may use it. However, if we want to boot a RT UOS. The RT UOS will need TPM for security. In this case, we may can't unbind the TPM from SOS and pass it through to RT UOS.

We can disable TPM for SOS in
https://github.com/intel/meta-acrn/blob/master/recipes-kernel/linux/files/sos.cfg

(From your original patch)
# CONFIG_TCG_TPM is not set
# CONFIG_HW_RANDOM_TPM is not set
# CONFIG_TCG_CRB is not set

Reply:
Thx a lot.